packages/mermaid/src/diagrams/mindmap/svgDraw.ts

```diff
@@ -1,55 +1,18 @@
-import { select } from 'd3';
-import * as db from './mindmapDb.js';
+import type { D3Element } from '../../mermaidAPI.js';
 import { createText } from '../../rendering-util/createText.js';
+import type { FilledMindMapNode, MindmapDB } from './mindmapTypes.js';
+import type { MermaidConfigWithDefaults } from '../../config.js';
+import type { Point } from '../../types.js';
 const MAX_SECTIONS = 12;
 
-/**
- * @param {string} text The text to be wrapped
- * @param {number} width The max width of the text
- */
-function wrap(text, width) {
-  text.each(function () {
-    var text = select(this),
-      words = text
-        .text()
-        .split(/(\s+|<br\/>)/)
-        .reverse(),
-      word,
-      line = [],
-      lineHeight = 1.1, // ems
-      y = text.attr('y'),
-      dy = parseFloat(text.attr('dy')),
-      tspan = text
-        .text(null)
-        .append('tspan')
-        .attr('x', 0)
-        .attr('y', y)
-        .attr('dy', dy + 'em');
-    for (let j = 0; j < words.length; j++) {
-      word = words[words.length - 1 - j];
-      line.push(word);
-      tspan.text(line.join(' ').trim());
-      if (tspan.node().getComputedTextLength() > width || word === '<br/>') {
-        line.pop();
-        tspan.text(line.join(' ').trim());
-        if (word === '<br/>') {
-          line = [''];
-        } else {
-          line = [word];
-        }
+type ShapeFunction = (
+  db: MindmapDB,
+  elem: D3Element,
+  node: FilledMindMapNode,
+  section?: number
+) => void;
```

Comment:
Wouldn't be better to move this into `mindmapTypes.ts`?

Reply:
I don't think so. This type is not used across files, and is not a generic type, there is no need to move it out.

Reply:
Okay.